cmd/derper/derper.go

```diff
@@ -327,6 +327,9 @@ func main() {
 
 // For captive portal detection
 func serveNoContent(w http.ResponseWriter, r *http.Request) {
+	if challenge := r.Header.Get("X-Tailscale-Challenge"); challenge != "" {
```

Comment:
validate that it is only ascii non-whitespace? or only hex maybe? and not too long?

Reply:
Sure, easy enough - no reason to allow tons of data or strange characters here. Added a test as well, because I don't add constraints without testing them 😃